### PR TITLE
alttab: init at 1.4.0

### PR DESCRIPTION
**Pull request by sgraf812**
###### Motivation for this change

`alttab` is a window switcher for X11, designed for minimalistic window
managers or a standalone X11 session.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/alttab/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, pkgconfig, ronn, libpng, uthash
+, xorg }:
+
+stdenv.mkDerivation rec {
+  version = "1.4.0";
+
+  name = "alttab-${version}";
```
Comment (alexarice) [inline pkgs/tools/X11/alttab/default.nix]: Could you use `pname = "alttab"` instead

File: pkgs/tools/X11/alttab/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sagb";
+    repo = name;
```
Comment (alexarice) [inline pkgs/tools/X11/alttab/default.nix]: This doesn't build for me, I think the version should not be in the repo name. Changing to `repo = "alttab"` makes this work for me

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -645,6 +645,8 @@ in
 
   altermime = callPackage ../tools/networking/altermime {};
 
+  alttab = callPackage ../tools/X11/alttab { };
```
Comment (alexarice) [inline pkgs/top-level/all-packages.nix]: I feel this isn't the right place but the file is in such a mess I can't tell where the right place is

**Review by alexarice — COMMENTED** [review]
The commit should probably be titled "alttab: init at 1.4.0". I can get this to build with the changes I mentioned but can't test it as I do not run X11

File: pkgs/tools/X11/alttab/default.nix (hunk 2 above)
Comment (sgraf812) [reply pkgs/tools/X11/alttab/default.nix]: Ouch... Thanks for catching this! I changed the `name` to its current definition at some point... Apparently this didn't tried to fetch from GitHub again.

If I understand the `pname` mechanics right, it should be enough to set `pname = "alttab"` above and set `repo = pname` instead, right?

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (sgraf812) [reply pkgs/top-level/all-packages.nix]: I assumed the file was ordered alphabetically... With some exceptions anyway

Comment (alexarice) [reply pkgs/top-level/all-packages.nix]: Its meant to be in sections and then alphabetically in those sections I think

File: pkgs/tools/X11/alttab/default.nix (hunk 2 above)
Comment (alexarice) [reply pkgs/tools/X11/alttab/default.nix]: that should work

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (alexarice) [reply pkgs/top-level/all-packages.nix]: I wouldn't worry about it for now

**Comment by sgraf812** [comment]
Thanks for your review @alexarice! I amended some changes and built it on another machine, so it should just work now.

**Review by c0bw3b — APPROVED** [review]
Tested locally and it works fine.

**Comment by c0bw3b** [comment]
Let's check build on Darwin and ARMv8:
@GrahamcOfBorg build alttab

**Pull request merged**